zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template_opts.rs

```diff
@@ -0,0 +1,92 @@
+//! Parameter types for the `getblocktemplate` RPC.
+
+use super::hex_data::HexData;
+
+/// Defines whether the RPC method should generate a block template or attempt to validate a block proposal.
+/// `Proposal` mode is currently unsupported and will return an error.
+#[derive(Clone, Debug, serde::Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum GetBlockTemplateRequestMode {
+    /// Indicates a request for a block template.
+    Template,
+
+    /// Indicates a request to validate block data.
+    /// Currently unsupported and will return an error.
+    Proposal,
+}
+
+impl Default for GetBlockTemplateRequestMode {
+    fn default() -> Self {
+        Self::Template
+    }
+}
+
+/// Valid `capabilities` values that indicate client-side support.
+#[derive(Clone, Debug, serde::Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+#[allow(dead_code)]
+pub enum GetBlockTemplateCapability {
+    /// Long Polling support.
+    /// Currently ignored by zebra.
+    LongPoll,
+
+    /// Information for coinbase transaction, default template data with the `coinbasetxn` field.
+    /// Currently ignored by zebra.
+    CoinbaseTxn,
+
+    /// Coinbase value, template response provides a `coinbasevalue` field and omits `coinbasetxn` field.
+    /// Currently ignored by zebra.
+    CoinbaseValue,
+
+    /// Components of the coinbase transaction.
+    /// Currently ignored by zebra.
+    CoinbaseAux,
+
+    /// Currently ignored by zcashd and zebra.
+    Proposal,
+
+    /// Currently ignored by zcashd and zebra.
+    ServerList,
+
+    /// Currently ignored by zcashd and zebra.
+    WorkId,
+
+    /// Unknown capability to fill in for mutations.
+    // TODO: Fill out valid mutations capabilities.
+    #[serde(other)]
+    UnknownCapability,
```

Comment:
Does this need to be `UnknownCapability(String)` to store mutations like "time"?

I guess we'll find out when we test it!

```suggestion
    /// Unknown capability to fill in for mutations.
    // TODO: Fill out valid mutations capabilities.
    #[serde(other)]
    UnknownCapability(String),
```

Reply:
Oh, I saw your other comment, I guess the exact value doesn't matter if we're ignoring them anyway.